llama_index/embeddings/mutli_modal_base.py

```diff
@@ -0,0 +1,186 @@
+                },
+            )
+        return image_embedding
+
+    async def aget_image_embedding(self, img_file_path: str) -> Embedding:
+        """Get image embedding."""
+        with self.callback_manager.event(
+            CBEventType.EMBEDDING, payload={EventPayload.SERIALIZED: self.to_dict()}
+        ) as event:
+            image_embedding = await self._aget_image_embedding(img_file_path)
+
+            event.on_end(
+                payload={
+                    EventPayload.CHUNKS: [img_file_path],
+                    EventPayload.EMBEDDINGS: [image_embedding],
+                },
+            )
+        return image_embedding
+
+    def _get_image_embeddings(self, img_file_paths: List[str]) -> List[Embedding]:
+        """
+        Embed the input sequence of image synchronously.
+
+        Subclasses can implement this method if batch queries are supported.
+        """
+        # Default implementation just loops over _get_image_embedding
+        return [
+            self._get_image_embedding(img_file_path) for img_file_path in img_file_paths
+        ]
+
+    async def _aget_image_embeddings(
+        self, img_file_paths: List[str]
+    ) -> List[Embedding]:
+        """
+        Embed the input sequence of image asynchronously.
+
+        Subclasses can implement this method if batch queries are supported.
+        """
+        return await asyncio.gather(
+            *[
+                self._aget_image_embedding(img_file_path)
+                for img_file_path in img_file_paths
+            ]
+        )
+
+    def get_image_embedding_batch(
+        self, img_file_paths: List[str], show_progress: bool = False
+    ) -> List[Embedding]:
+        """Get a list of image embeddings, with batching."""
+        cur_batch: List[str] = []
+        result_embeddings: List[Embedding] = []
+
+        queue_with_progress = enumerate(
+            get_tqdm_iterable(img_file_paths, show_progress, "Generating embeddings")
+        )
+
+        for idx, img_file_path in queue_with_progress:
+            cur_batch.append(img_file_path)
+            if (
+                idx == len(img_file_paths) - 1
+                or len(cur_batch) == self.embed_batch_size
+            ):
+                # flush
+                with self.callback_manager.event(
+                    CBEventType.EMBEDDING,
+                    payload={EventPayload.SERIALIZED: self.to_dict()},
+                ) as event:
+                    embeddings = self._get_image_embeddings(cur_batch)
+                    result_embeddings.extend(embeddings)
+                    event.on_end(
+                        payload={
+                            EventPayload.CHUNKS: cur_batch,
+                            EventPayload.EMBEDDINGS: embeddings,
+                        },
+                    )
+                cur_batch = []
+
+        return result_embeddings
+
+    async def aget_image_embedding_batch(
```

Comment:
Hmm, I wonder if this will actually work with a local model running? 

I think it might actually end up running sequentially since, but that's ok lol just a caveat

Reply:
I think the base embeddings have the same issue

Reply:
yes @logan-markewich this is a local clip model
is it neccessary to make a async call in this case
I just make it sequentially